src/consumers.ts

```diff
@@ -49,6 +52,139 @@ interface IPipeline<T> {
 /* Diagnostics producer type */
 type DiagnosticProducer = IProducer<Diagnostic[]>;
 
+/* Package vulnerability data */
+class Package
+{
+    name: string;
+    version: string;
+    packageCount: number;
+    vulnerabilityCount: number;
+    advisoryCount: number;
+    exploitCount: number;
+    highestSeverity: string;
+    recommendedVersion: string;
+    range: Range;
+
+    constructor(
+        name: string, version: string, packageCount: number,
+        vulnerabilityCount: number, advisoryCount: number,
+        exploitCount: number, highestSeverity: string,
+        recommendedVersion: string, range: Range) {
+            this.name = name;
+            this.version = version;
+            this.packageCount = packageCount || 0;
+            this.vulnerabilityCount = vulnerabilityCount || 0;
+            this.advisoryCount = advisoryCount || 0;
+            this.exploitCount = exploitCount || 0;
+            this.highestSeverity = highestSeverity;
+            this.recommendedVersion = recommendedVersion;
+            this.range = range
+    }
+
+    getDiagnostic(): Diagnostic {
+        /* The diagnostic's severity. */
+        let diagSeverity: any;
+
+        if (this.vulnerabilityCount == 0 && this.advisoryCount > 0) {
+            diagSeverity = DiagnosticSeverity.Information;
+        } else {
+            diagSeverity = DiagnosticSeverity.Error;
+        }
+
+        const recommendedVersion = this.recommendedVersion || "N/A";
+        const exploitCount = this.exploitCount || "unavailable";
+        const msg = `${this.name}: ${this.version}
+Number of packages: ${this.packageCount}
+Known security vulnerability: ${this.vulnerabilityCount}
+Security advisory: ${this.advisoryCount}
+Exploits: ${exploitCount}
+Highest severity: ${this.highestSeverity}
+Recommendation: ${recommendedVersion}`;
+
+        return {
+            severity: diagSeverity,
+            range: this.range,
+            message: msg,
+            source: '\nDependency Analytics Plugin [Powered by Snyk]',
+        };
+    }
+}
+
+/* Package aggregator class */
+class PackageAggregator
```

Comment:
It is more of a VulnerabilityAggregator than Package

Reply:
As commented before, "package" is apt name for data structure that we are using, and this aggregator actually combines / merges this packages together. Its not only vulnerability, but it also merges recommended version and exploits etc. Not renaming right now.

Reply:
Accepted and done.